packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -199,6 +215,24 @@ moduleFor('Components test: curly components', class extends RenderingTest {
     this.assertComponentElement(this.firstChild, { tagName: 'div', attrs: { class: 'ember-view' }, content: 'hello' });
   }
 
+  ['@test class property on components can be dynamic']() {
+    this.registerComponent('foo-bar', { template: 'hello' });
+
+    this.render('{{foo-bar class=(if fooBar "foo-bar")}}', {
+      fooBar: true
+    });
+
+    this.assertComponentElement(this.firstChild, { content: 'hello', attrs: { 'class': classes('ember-view foo-bar') } });
+
+    this.runTask(() => this.rerender());
+
+    this.assertComponentElement(this.firstChild, { content: 'hello', attrs: { 'class': classes('foo-bar ember-view') } });
```

Comment:
Although, IIRC, we are sorting in the test helper to do a stable comparison.  Maybe this is the result of that?  If so, we should probably test the real element order too somehow...